at_client/connections/atmonitorconnection.py

```diff
@@ -113,11 +113,11 @@ def stop_monitor(self):
         self._last_heartbeat_sent_time = self._last_heartbeat_ack_time = TimeUtil.current_time_millis()
         self.disconnect()
         
-    def _run(self):
+    def _run(self, regex):
         what = ""
         first = True
         try:
-            monitor_cmd = "monitor:" + str(self.last_received_time)
+            monitor_cmd = "monitor " + regex
```

Comment:
```suggestion
            monitor_cmd = "monitor:" + str(self.last_received_time) + " " + regex
```


Check [MonitorVerbBuilder](https://github.com/atsign-foundation/at_libraries/blob/trunk/packages/at_commons/lib/src/verb/monitor_verb_builder.dart) from `at_commons` lib

You can also use `format` if you find it more beautiful: 
`monitor_cmd = "monitor: {} {}".format(self.last_received_time, regex)`

Reply:
I've been messing around with last_received_time, out of curiosity what's the purpose of it? (I'm not quite sure how it works, is it necessary?)

Reply:
From `MonitorVerbBuilder`:
```
  The timestamp, in milliseconds since epoch, to be used when building the monitor command.
  When [lastNotificationTime] is supplied, server will only send notifications received at
  or after that timestamp
```

I think that the idea is to try not to receive all the notifications available. For instance, if connection gets lost and the application has to recconect, it would be a bit annoying that all the notifications were received again (fortunately, `last_received_time` will be sent and we will skip all previous ones). 

Also there's a mechanism to stop old notifications from being received (like "marking them as read" I guess), and even though I've not dived into that, maybe it has some relation with that mechanism and may be necessary in the future. 

> is it necessary?

It probably is. I didn't spend more time than to understand its meaning beyond what I mentioned earlier, but I guess that if it's on `Dart SDK` and also  `Java SDK`  it must be useful! 😄 

For now we know that it helps us in case of losing the connection, that is something.

Reply:
That's what I thought it was for, but I realize now where the issue was. I've been getting _literally_ all of the notifications to every exist and I wasn't sure if it was how I was handling things, but I think I've got it. 

Upon the running of the command the `last_recieved_time` is only initialized to 0, and only later (after the command has been executed) will the last_recieved_time get set to the notification epoch or currentTime.

Also I think the default monitor handles it all this stuff for you, just at a glance. ( not 100% sure, I'll have to look into it)